day_two_guide/topics/restore_etcd.adoc

```diff
@@ -0,0 +1,382 @@
++
+----
+# vi /etc/etc/etcd.conf
+----
+
+. Once those values are replaced, start the etcd service in the node joining
+the cluster:
++
+----
+# systemctl start etcd.service
+----
+
+. Check for error messages:
++
+----
+$ journalctl -fu etcd.service
+----
+
+. Repeat the above for every etcd node joining the cluster.
+
+. Verify the cluster status and cluster health once all the nodes joined:
++
+----
+# etcdctl2 member list
+5cd050b4d701: name=master1 peerURLs=https://10.0.0.7:2380 clientURLs=https://10.0.0.7:2379 isLeader=true
+d0c57659d8990cbd: name=master2 peerURLs=https://10.0.0.5:2380 clientURLs=https://10.0.0.5:2379 isLeader=false
+e4696d637de3eb2d: name=master3 peerURLs=https://10.0.0.6:2380 clientURLs=https://10.0.0.6:2379 isLeader=false
+----
++
+----
+# etcdctl2 cluster-health
+member 5cd050b4d701 is healthy: got healthy result from https://10.0.0.7:2379
+member d0c57659d8990cbd is healthy: got healthy result from https://10.0.0.5:2379
+member e4696d637de3eb2d is healthy: got healthy result from https://10.0.0.6:2379
+cluster is healthy
+----
+
+==== Restoring etcd for v3
+
+The restore procedure for v3 data is similar to the v2 data.
+
+Snapshot integrity may be optionally verified at restore time. If the snapshot
+is taken with `etcdctl snapshot save`, it will have an integrity hash that is
+checked by `etcdctl snapshot restore`. If the snapshot is copied from the data
+directory, there is no integrity hash and it will only restore by using
+`--skip-hash-check`.
+
+[IMPORTANT]
+====
+The procedure to restore only the v3 data must be performed on a single etcd
+host. You can then add the rest of the nodes to the cluster.
+====
+
+[discrete]
+== Procedure
+
+. Stop all etcd services:
++
+----
+# systemctl stop etcd.service
+----
+
+. Clear all old data, because `etcdctl` recreates it in the node where the
+restore procedure is going to be performed:
++
+----
+# rm -Rf /var/lib/etcd
+----
+
+. Use the `snapshot restore` command with the data from `/etc/etcd/etcd.conf` to
+match the following command:
++
+----
+# etcdctl3 snapshot restore /backup/etcd-xxxxxx/backup.db \
+  --data-dir /var/lib/etcd \
+  --name master-0.example.com \
+  --initial-cluster "master-0.example.com=https://192.168.55.8:2380" \ --initial-cluster-token "etcd-cluster-1" \
+  --initial-advertise-peer-urls https://192.168.55.8:2380
+
+2017-10-03 08:55:32.440779 I | mvcc: restore compact to 1041269
+2017-10-03 08:55:32.468244 I | etcdserver/membership: added member 40bef1f6c79b3163 [https://192.168.55.8:2380] to cluster 26841ebcf610583c
+----
+
+. Restore permissions and `selinux` context to the restored files:
++
+----
+# chown -R etcd.etcd /var/lib/etcd/
+# restorecon -Rv /var/lib/etcd
+----
+
+. Start the etcd service:
++
+----
+# systemctl start etcd
+----
+
+. Check for any error messages:
++
+----
+$ journalctl -fu etcd.service
+----
+
+*Adding more nodes*
+
+Once the first instance is running, it is safe to restore multiple etcd servers as desired.
```

Comment:
s/Once/After
s/it is safe to restore multiple etcd servers as desired./you can add more etcd nodes to your cluster.